### PR TITLE
Add eslintrc with typescript support

### PR DESCRIPTION
**Pull request by alizhdanov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,11 @@
     "configuration"
   ],
   "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^2.8.0",
+    "@typescript-eslint/parser": "^2.8.0",
+    "eslint-config-airbnb-typescript": "^6.3.0",
+    "eslint-config-prettier": "^6.7.0",
+    "eslint-plugin-prettier": "^3.1.1",
```
Comment (yakovlevyuri) [inline package.json]: add `eslint-plugin-react-hooks`

File: common/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "plugins": ["@typescript-eslint"],
+  "rules": {
+    "import/prefer-default-export": "off",
+    "@typescript-eslint/ban-ts-ignore": "off",
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "no-shadow": "off",
```
Comment (jaylonez) [inline common/.eslintrc]: I would keep no-shadow for code clarity

File: common/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "plugin:prettier/recommended"
+  ],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "plugins": ["@typescript-eslint"],
+  "rules": {
+    "import/prefer-default-export": "off",
```
Comment (yakovlevyuri) [inline common/.eslintrc]: add 
```
 'react-hooks/rules-of-hooks': 'error',
 'react-hooks/exhaustive-deps': 'warn',
```

File: common/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    "@typescript-eslint/ban-ts-ignore": "off",
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "no-shadow": "off",
+    "@typescript-eslint/no-use-before-define": "off",
```
Comment (jaylonez) [inline common/.eslintrc]: What's the problem with no-use-before-define?

File: common/.eslintrc (hunk 3 above)
Comment (alizhdanov) [reply common/.eslintrc]: This is "base" config. For fe - check `frontend` folder

File: package.json (hunk 1 above)
Comment (alizhdanov) [reply package.json]: thanks

File: common/.eslintrc (hunk 4 above)
Comment (alizhdanov) [reply common/.eslintrc]: Changed rule a bit so it'd [work correctly with functions and classes](https://eslint.org/docs/rules/no-use-before-define#options). Good catch. Thanks!

**Comment by alizhdanov** [comment]
Closing because a bit outdated, up to date version -> https://github.com/oakslab/ooo-config-examples/pull/12

**Pull request closed without merging**